src/Microsoft.ML.Data/Scorers/PredictionTransformer.cs

```diff
@@ -91,12 +97,27 @@ private protected PredictionTransformerBase(IHost host, ModelLoadContext ctx)
 
             // *** Binary format ***
             // model: prediction model.
-            // stream: empty data view that contains train schema.
-            // id of string: feature column.
 
             ctx.LoadModel<TModel, SignatureLoadModel>(host, out TModel model, DirModel);
             Model = model;
 
+            InitializeLogic(host, ctx);
+        }
+
+        [BestFriend]
+        private protected PredictionTransformerBase(IHost host, ModelLoadContext ctx, TModel model)
+        {
+            Host = host;
+            Model = model; // prediction model
+            InitializeLogic(host, ctx);
+        }
+
+        private protected void InitializeLogic(IHost host, ModelLoadContext ctx)
```

Comment:
(minor) This (and other InitializeLogic methods) should just be `private`. No one outside of this class needs to call it.

Reply:
Now fixed in my latest commit.